Changelog: Mention some more changes since 1.8.0 (#923)
simplepie/simplepie: CHANGELOG.md

### Added

- New method `SimplePie\SimplePie::set_http_client()` for providing PSR-18 HTTP client and PSR-17 factories (by @Art4 in [#777](https://github.com/simplepie/simplepie/pull/777))
- New method `SimplePie\Locator::set_http_client()` for providing PSR-18 HTTP client and PSR-17 factories (by @Art4 in [#777](https://github.com/simplepie/simplepie/pull/777))
- New method `SimplePie\Sanitize::set_http_client()` for providing PSR-18 HTTP client and PSR-17 factories (by @Art4 in [#777](https://github.com/simplepie/simplepie/pull/777))

### Changed

- Use `idn_to_ascii` function instead of `idna_convert` library (requires `intl` extension or a [polyfill](https://github.com/symfony/polyfill-intl-idn)) (by @jtojnar in [#785](https://github.com/simplepie/simplepie/pull/785))

> - Improve precision of type annotations and increased level of PHPStan checks (by @jtojnar in [#837](https://github.com/simplepie/simplepie/pull/837))

Reviewer: Maybe we should mention that `declare(strict_types=1);` are declared furthermore in contrast to the removal in 1.8.1?

Author: That was already done in 1.8.0?

Reviewer: Yes, see #763

- Use native `gzdecode` function instead of internal PHP implementation (by @jtojnar in [#882](https://github.com/simplepie/simplepie/pull/882))

> - `SimplePie::set_curl_options([CURLOPT_HTTPHEADER => …])` will no longer prevent the following headers defined by SimplePie from being sent `Accept`, `if-modified-since`, `if-none-match` and `X-FORWARDED-FOR`. (by @Alkarex in [#912](https://github.com/simplepie/simplepie/pull/912))

Reviewer: ```suggestion
- `SimplePie\SimplePie::set_curl_options([CURLOPT_HTTPHEADER => …])` will no longer prevent the following headers defined by SimplePie from being sent `Accept`, `if-modified-since`, `if-none-match` and `X-FORWARDED-FOR`. (by @Alkarex in [#912](https://github.com/simplepie/simplepie/pull/912))
```

### Fixed

- Do not crash when `Content-Type` header is missing (by @Art4 in [#774](https://github.com/simplepie/simplepie/pull/774))

> - Correct `File::$permanent_url` when multiple redirects are chained (by @jtojnar in [#812](https://github.com/simplepie/simplepie/pull/812))

Reviewer: ```suggestion
- Correct `SijmplePie\File::$permanent_url` when multiple redirects are chained (by @jtojnar in [#812](https://github.com/simplepie/simplepie/pull/812))
```

> - Improve PHP 8.4 compatibility (by @jrfnl in [#880](https://github.com/simplepie/simplepie/pull/880), [#881](https://github.com/simplepie/simplepie/pull/881), @jtojnar in [#837](https://github.com/simplepie/simplepie/pull/837), @Girgias in [#851](https://github.com/simplepie/simplepie/pull/851))

Reviewer: This was backported to 1.8.1

- Fix regression from 1.8.0 in cache invalidation (by @Alkarex in [#883](https://github.com/simplepie/simplepie/pull/883))

> - `SimplePie::strip_htmltags()` correctly propagates `$encode` argument to `Sanitize` (by @jtojnar in [#894](https://github.com/simplepie/simplepie/pull/894))

Reviewer: This change was backported into 1.8.1.

```suggestion
- `SimplePie\SimplePie::strip_htmltags()` correctly propagates `$encode` argument to `Sanitize` (by @jtojnar in [#894](https://github.com/simplepie/simplepie/pull/894))
```

> - Stop `File` from mangling UTF-16 and UTF-32 encoded feeds (by @Alkarex in [#916](https://github.com/simplepie/simplepie/pull/916) and [#917](https://github.com/simplepie/simplepie/pull/917))

Reviewer: ```suggestion
- Stop `SimplePie\File` from mangling UTF-16 and UTF-32 encoded feeds (by @Alkarex in [#916](https://github.com/simplepie/simplepie/pull/916) and [#917](https://github.com/simplepie/simplepie/pull/917))
```

### Removed

- Third-party `idna_convert` library is no longer part of the repository (by @jtojnar in [#785](https://github.com/simplepie/simplepie/pull/785))

### Deprecated

> - Most public properties on `SimplePie\File` are deprecated in favour of new methods of `SimplePie\Response` interface:

Reviewer: The `SimplePie\HTTP\Response` interface is marked as internal. We should not reference it in the CHANGELOG, so we can have some changes in the interface before it will be non-internal in 1.10.

```suggestion
- Most public properties on `SimplePie\File` are deprecated in favour of new methods:
```

Author: Hmm, good point. Though that means the suggested replacement methods should not be mentioned either?

>   - `SimplePie\File::$url` → `SimplePie\File::get_final_requested_uri()`
>   - `SimplePie\File::$useragent` → Pass the user agent in constructor.
>   - `SimplePie\File::$body` → `SimplePie\File::get_body_content()`
>   - `SimplePie\File::$headers` → `SimplePie\File::get_headers()`
>   - `SimplePie\File::$status_code` → `SimplePie\File::get_status_code()`
>   - `SimplePie\File::$method` – no replacement: backend is implementation detail which you should not care about; to see if the file was retrieved over HTTP, check the protocol of `SimplePie\File::get_final_requested_uri()`.
>   - `SimplePie\File::$permanent_url` → `SimplePie\File::get_permanent_uri()`

Reviewer: ```suggestion
  - `SimplePie\File::$url`, use `SimplePie\File::get_final_requested_uri()` instead.
  - `SimplePie\File::$useragent`, pass the user agent in constructor instead.
  - `SimplePie\File::$body`, use `SimplePie\File::get_body_content()` instead.
  - `SimplePie\File::$headers`, use `SimplePie\File::get_headers()` instead.
  - `SimplePie\File::$status_code`, use `SimplePie\File::get_status_code()` instead.
  - `SimplePie\File::$method` witout replacement: backend is implementation detail which you should not care about; to see if the file was retrieved over HTTP, check the protocol of `SimplePie\File::get_final_requested_uri()`.
  - `SimplePie\File::$permanent_url`, use `SimplePie\File::get_permanent_uri()` instead.
```
